components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx

```diff
@@ -37,111 +38,71 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
         toggleEmbedVisibility: PropTypes.func.isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * The function to get open graph data for a link
-             */
-            getOpenGraphMetadata: PropTypes.func.isRequired,
-        }).isRequired,
     }
 
     constructor(props) {
         super(props);
-        this.largeImageMinWidth = 150;
-        this.imageDimentions = { // Image dimentions in pixels.
-            height: 80,
-            width: 80,
-        };
-        this.textMaxLength = 300;
-        this.textEllipsis = '...';
-        this.largeImageMinRatio = 16 / 9;
-        this.smallImageContainerLeftPadding = 15;
 
-        this.imageRatio = null;
-
-        this.smallImageContainer = null;
-        this.smallImageElement = null;
-
-        this.IMAGE_LOADED = {
-            LOADING: 'loading',
-            YES: 'yes',
-            ERROR: 'error',
-        };
-
-        this.fetchData = this.fetchData.bind(this);
-        this.onImageLoad = this.onImageLoad.bind(this);
-        this.onImageError = this.onImageError.bind(this);
         this.handleRemovePreview = this.handleRemovePreview.bind(this);
-    }
 
-    UNSAFE_componentWillMount() { // eslint-disable-line camelcase
-        const removePreview = this.isRemovePreview(this.props.post, this.props.currentUser);
-
-        this.setState({
-            imageLoaded: this.IMAGE_LOADED.LOADING,
-            hasLargeImage: false,
+        const removePreview = this.isRemovePreview(props.post, props.currentUser);
+        const imageUrl = this.getBestImageUrl(props.openGraphData);
+        const hasLargeImage = this.hasLargeImage(props.post.metadata.images[imageUrl]);
+        this.state = {
+            hasLargeImage,
             removePreview,
-        });
-        this.fetchData(this.props.link);
+            imageUrl,
+        };
     }
 
     UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
         if (!Utils.areObjectsEqual(nextProps.post, this.props.post)) {
             const removePreview = this.isRemovePreview(nextProps.post, nextProps.currentUser);
+            const imageUrl = this.getBestImageUrl(nextProps.openGraphData);
+            const hasLargeImage = this.hasLargeImage(nextProps.post.metadata.images[imageUrl]);
+
             this.setState({
+                hasLargeImage,
                 removePreview,
+                imageUrl,
             });
         }
-        if (nextProps.link !== this.props.link) {
-            this.fetchData(nextProps.link);
-        }
     }
 
     componentDidUpdate() {
         setTimeout(postListScrollChange, 0);
     }
 
-    fetchData(url) {
-        if (!this.props.openGraphData) {
-            this.props.actions.getOpenGraphMetadata(url);
-        }
-    }
-
     getBestImageUrl(data) {
         if (Utils.isEmptyObject(data.images)) {
             return null;
         }
+        const imageDimentions = { // Image dimentions in pixels.
```

Comment:
maybe this can be a constant so we don't need to create this object every time the function gets called?

Reply:
I guess you meant defining it outside the functions somewhere in constructor or at top of file? 
Should not be a problem as this does not effect render cycles. function should eventually anyway return a new result as it is not a selector.